Zend/tests/final_properties/access/property_assignment_combined1.phpt

```diff
@@ -0,0 +1,106 @@
+--TEST--
+Test that final properties can't be mutated by combined assignment operators
+--FILE--
+<?php
+
+class Foo
+{
+    final public int $property1 = 1;
+    final public ?string $property2;
+}
+
+$foo = new Foo();
+
+try {
+    $foo->property1 += 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 -= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 *= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 /= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 **= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 &= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 |= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 ^= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 <<= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 >>= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+$foo->property2 ??= null;
```

Comment:
Could document in the source of the test case that this is the first assignment to property2, and the test chose `null` because `??=` would do nothing if the property value was not null and not initialized - I assume others may be as confused as you.

Same for the tests of static.

Reply:
Will do! Actually, I was already considering it before, but then changed my mind.